buffer/src/main/java/io/netty/buffer/ByteBuf.java

```diff
@@ -2010,6 +2044,20 @@
     public abstract ByteBuf slice();
 
     /**
+     * Returns a retained slice of this buffer's readable bytes. Modifying the content
+     * of the returned buffer or this buffer affects each other's content
+     * while they maintain separate indexes and marks.  This method is
+     * identical to {@code buf.slice(buf.readerIndex(), buf.readableBytes())}.
+     * This method does not modify {@code readerIndex} or {@code writerIndex} of
+     * this buffer.
+     * <p>
+     * Note that this method returns a {@linkplain #retain() retained} buffer unlike {@link #slice()}.
+     * This method behaves similarly to {@code slice().retain()} except that this method may return
+     * a buffer implementation that produces less garbage.
+     */
+    public abstract ByteBuf rslice();
```

Comment:
What about: `rslice` -> `sliceAndRetain()`

I agree with your comment https://github.com/netty/netty/pull/5144#issuecomment-210221554

> wanted the method names start with slice/duplicate so that they are grouped together with slice() and duplicate()

The name may be a bit verbose but it enhances discoverability and understandability IMO. This method is inherently a compound operation so explicitly listing the operations combined with a `And` clearly communicates what the method should do.